### PR TITLE
Fix remote vehicles not updating in garage

### LINKED ISSUE #558: Garage - Unable to edit non-local vehicles
**Arma 3 Version:** `2.00` (stable)
**CBA Version:** `3.15.2` (stable)
**ZEN Version:** `1.9.0` (stable / dev + commit hash)

**Mods:**
```
- CBA_A3
- ZEN
```

**Description:**
Garage functionality in ZEN does not work for non-local vehicles. Setting locality to yourself is a current workaround.

**Expected behavior:**
Ability to edit vehicle in Garage even if it's not local to me (eg. on server or another Zeus).

**Steps to reproduce:**
1. Spawn vehicle
2. Set owner to server
3. Attempt to edit via Garage

**Where did the issue occur?**
- Dedicated

**Additional information:**
[`BIS_fnc_initVehicle`](https://community.bistudio.com/wiki/BIS_fnc_initVehicle) requires local arguments

**RPT log file:**
- /

**Screenshots:**
/

### PR DESCRIPTION
**Pull request by neilzar**
**When merged this pull request will:**
- Fixes #558

### CODE REVIEW

File: addons/garage/functions/fnc_applyToAll.sqf [hunk 1]
```diff
@@ -22,6 +22,6 @@ private _vehicleType = typeOf GVAR(center);
 
 {
     if (typeOf _x isEqualTo _vehicleType) then {
-        [_x, _texture, _animations, true] call BIS_fnc_initVehicle;
+        [QEGVAR(common,initVehicle), [_x, _texture, _animations, true], _x] call cba_fnc_targetEvent;
```
Comment (CreepPork) [inline addons/garage/functions/fnc_applyToAll.sqf:25]:
```suggestion
        [QEGVAR(common,initVehicle), [_x, _texture, _animations, true], _x] call CBA_fnc_targetEvent;
```

File: addons/garage/functions/fnc_onAnimationSelect.sqf [hunk 2]
```diff
@@ -31,4 +31,5 @@ for "_i" from 0 to (lbSize _ctrlListAnimations - 1) do {
 };
 
 // Update vehicle animations
-[GVAR(center), nil, _animations, true] call BIS_fnc_initVehicle;
+private _vehicle = GVAR(center);
+[QEGVAR(common,initVehicle), [_vehicle, nil, _animations, true], _vehicle] call cba_fnc_targetEvent;
```
Comment (CreepPork) [inline addons/garage/functions/fnc_onAnimationSelect.sqf:35]:
```suggestion
[QEGVAR(common,initVehicle), [_vehicle, nil, _animations, true], _vehicle] call CBA_fnc_targetEvent;
```

File: addons/garage/functions/fnc_onTextureSelect.sqf [hunk 3]
```diff
@@ -27,4 +27,5 @@ for "_i" from 0 to (lbSize _ctrlListTextures - 1) do {
 _ctrlListTextures lbSetPicture [_selectedIndex, ICON_CHECKED];
 
 // Update vehicle textures
-[GVAR(center), [_ctrlListTextures lbData _selectedIndex, 1]] call BIS_fnc_initVehicle;
+private _vehicle = GVAR(center);
+[QEGVAR(common,initVehicle), [_vehicle, [_ctrlListTextures lbData _selectedIndex, 1]], _vehicle] call cba_fnc_targetEvent;
```
Comment (CreepPork) [inline addons/garage/functions/fnc_onTextureSelect.sqf:31]:
```suggestion
[QEGVAR(common,initVehicle), [_vehicle, [_ctrlListTextures lbData _selectedIndex, 1]], _vehicle] call CBA_fnc_targetEvent;
```

**Review by CreepPork — APPROVED** [review]

File: addons/common/XEH_postInit.sqf [hunk 4]
```diff
@@ -283,6 +283,10 @@
     _vehicle setVehicleReportOwnPosition _mode;
 }] call CBA_fnc_addEventHandler;
 
+[QGVAR(initVehicle), {
+    _this call BIS_fnc_initVehicle;
+}] call CBA_fnc_addEventHandler;
```
Comment (mharis001) [inline addons/common/XEH_postInit.sqf:288]: Don't need `_this call ...`, can have `BIS_fnc_initVehicle` directly.

File: addons/garage/functions/fnc_onAnimationSelect.sqf [hunk 5]
```diff
@@ -31,4 +31,5 @@ for "_i" from 0 to (lbSize _ctrlListAnimations - 1) do {
 };
 
 // Update vehicle animations
-[GVAR(center), nil, _animations, true] call BIS_fnc_initVehicle;
+private _vehicle = GVAR(center);
+[QEGVAR(common,initVehicle), [_vehicle, nil, _animations, true], _vehicle] call CBA_fnc_targetEvent;
```
Comment (mharis001) [inline addons/garage/functions/fnc_onAnimationSelect.sqf:35]: For just two uses, I think it would be fine to use the global variable directly.

**Review by mharis001 — APPROVED** [review]

**Pull request merged**